spec/view_objects/allocations_view_spec.rb

```diff
@@ -0,0 +1,368 @@
+                                   status: AllocationsView::Status::REQUESTED }])
+        end
+
+        it "does not display number of requested places in the status" do
+          expect(subject.first[:status]).to eq("REQUESTED")
+        end
+      end
+    end
+
+    describe "#not_requested_allocations" do
+      subject { AllocationsView.new(training_providers: training_providers, allocations: allocations).not_requested_allocations_statuses }
+
+      context "returns allocations where status is 'NOT REQUESTED'" do
+        let(:declined_allocation) { build(:allocation, :declined, accredited_body: accredited_body, provider: training_provider, number_of_places: 0) }
+
+        let(:training_providers) { [training_provider] }
+
+        let(:allocations) { [declined_allocation] }
+
+        it {
+          expect(subject).to eq([{
+            training_provider_name: training_provider.provider_name,
+            training_provider_code: training_provider.provider_code,
+            status: AllocationsView::Status::NOT_REQUESTED,
+            status_colour: AllocationsView::Colour::RED,
+          }])
+        }
+      end
+
+      context "returns allocations where status is 'YET TO REQUEST'" do
+        let(:allocations) { [] }
+
+        let(:training_providers) { [training_provider] }
+
+        it {
+          expect(subject).to eq([{
+            training_provider_name: training_provider.provider_name,
+            training_provider_code: training_provider.provider_code,
+            status: AllocationsView::Status::NO_REQUEST_SENT,
+            status_colour: AllocationsView::Colour::GREY,
+          }])
+        }
+      end
+
+      context "no allocations are returned if their status is 'REQUESTED" do
+        let(:repeat_allocation) do
+          build(:allocation, :repeat, accredited_body: accredited_body, provider: training_provider, number_of_places: 1)
+        end
+
+        let(:allocations) { [repeat_allocation] }
+
+        let(:training_providers) { [training_provider] }
+
+        it {
+          expect(subject).to eq([])
+        }
+      end
+
+      context "no allocations are returned if their status is 'INITIAL'" do
+        let(:initial_allocation) do
+          build(:allocation, :initial, accredited_body: accredited_body, provider: training_provider, number_of_places: 1)
+        end
+
+        let(:allocations) { [initial_allocation] }
+
+        let(:training_providers) { [training_provider] }
+
+        it {
+          expect(subject).to eq([])
+        }
+      end
+
+      describe "allocation ordering" do
+        before do
+          training_provider.provider_name = "Training Provider A"
+          another_training_provider.provider_name = "Training Provider B"
+        end
+
+        let(:allocations) { [] }
+
+        it "is sorted by the provider name" do
+          expect(subject).to eq([{ training_provider_name: training_provider.provider_name,
```

Comment:
Could we match the indentation of this scenario to the ones above

Reply:
done, through we should add a linter rule for that next time